packages/core/saved-objects/core-saved-objects-base-server-internal/src/saved_objects_config.ts

```diff
@@ -16,6 +16,7 @@ const migrationSchema = schema.object({
   }),
   batchSize: schema.number({ defaultValue: 1_000 }),
   maxBatchSizeBytes: schema.byteSize({ defaultValue: '100mb' }), // 100mb is the default http.max_content_length Elasticsearch config value
+  maxReadBatchSizeBytes: schema.byteSize({ defaultValue: 536870888, max: 536870888 }),
```

Comment:
IIUC this parameter has a default of `buffer.constants.MAX_STRING_LENGTH` already.
https://github.com/elastic/elastic-transport-js/blob/main/src/connection/HttpConnection.ts#L100
We're adding it here just for testing purposes, right?

Could we use the 'buffer.constants.MAX_STRING_LENGTH' instead of hardcoding the value here?

Reply:
I made it configurable because it was easier to test, but it's also a useful feature. On small memory instances buffer.constants.MAX_STRING_LENGTH could cause an OOM so even without testing it felt worth exposing this.

Agree about using the existing `buffer.constants.MAX_STRING_LENGTH` though 👍